addons/project/static/src/js/project.js

```diff
@@ -51,19 +51,23 @@ KanbanRecord.include({
         }
 
         function open_cover_images_dialog(attachment_ids) {
-            var cover_id = self.record.displayed_image_id.raw_value[0];
+            var cover_id = self.record.displayed_image_id && self.record.displayed_image_id.raw_value[0];
             var $content = $(QWeb.render("project.SetCoverModal", {
                 cover_id: cover_id,
                 attachment_ids: attachment_ids,
             }));
             var $imgs = $content.find('img');
-
             var dialog = new Dialog(self, {
                 title: _t("Set a Cover Image"),
                 buttons: [{text: _t("Select"), classes: 'btn-primary', close: true, disabled: !cover_id, click: function () {
-                    self._updateRecord({displayed_image_id: $imgs.filter('.o_selected').data('id')});
+                    var $img = $imgs.filter('.o_selected');
```

Comment:
@ged-odoo  we could not use `find` here because `$imgs` is list of images given by `var $imgs = $content.find('img');`